repos/system_upgrade/el7toel8/libraries/repofileutils.py

```diff
@@ -35,12 +35,26 @@ def parse_repofile(repofile):
     return RepositoryFile(file=repofile, data=data)
 
 
+def get_repodirs():
+    """
+    Return all directories yum scans for repository files.
+    By default, their paths on RHEL 7 should be:
+    ['/etc/yum.repos.d', '/etc/yum/repos.d', '/etc/distro.repos.d']
+
+    ATTENTION: Module dnf is reuired. The paths may not exist.
+    """
+    import dnf  # pylint: disable=import-outside-toplevel
```

Comment:
why do you need this import here?

Reply:
For `dnf.base` and `dnf.conf` in this function.
If I put it at the top level, Travis CI fails since a clean CentOS 7 doesn't have dnf.

Reply:
A more pythonic way of doing this is use try-except ImportError at the beginning of the module. This also removes the optional deps from the sys.modules, which makes other modules do not have access to it and other partially initialized modules.

```python
import warnings

try:
    import something_optional
except ImportError:
    warnings.warn('local_settings failed to import', ImportWarning)
```

Reply:
@ZhukovGreen Where do the warnings go?

Reply:
The snippet was generic. In our case it is probably api current logger warning @drehak

Reply:
DNF is available on CentOS 7. It's in `extras` repo.